python/lsst/pipe/tasks/drpDiaCalculationPipe.py

```diff
@@ -0,0 +1,139 @@
+# This file is part of pipe_tasks.
+
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+
+"""Pipeline for computing DiaObject summary/light curve values.
+"""
+
+__all__ = ["DrpDiaCalculationPipeTask",
+           "DrpDiaCalculationPipeConfig",
+           "DrpDiaCalculationPipeConnections"]
+
+import pandas as pd
+
+from lsst.meas.base import DiaObjectCalculationTask
+import lsst.pex.config as pexConfig
+import lsst.pipe.base as pipeBase
+
+
+class DrpDiaCalculationPipeConnections(pipeBase.PipelineTaskConnections,
+                                       dimensions=("tract", "patch", "skymap"),
+                                       defaultTemplates={"coaddName": "deep",
+                                                         "fakesType": ""}):
+    assocDiaSourceTable = pipeBase.connectionTypes.Input(
+        doc="Catalog of DiaSources covering the patch and associated with a "
+            "DiaObject.",
+        name="{fakesType}{coaddName}Diff_assocDiaSrcTable",
+        storageClass="DataFrame",
+        dimensions=("tract", "patch"),
+    )
+    diaObjectTable = pipeBase.connectionTypes.Input(
+        doc="Catalog of DiaObjects created from spatially associating "
+            "DiaSources.",
+        name="{fakesType}{coaddName}Diff_diaObjTable",
+        storageClass="DataFrame",
+        dimensions=("tract", "patch"),
+    )
+    fullDiaObjectTable = pipeBase.connectionTypes.Output(
+        doc="Catalog of DiaObjects created from spatially associating "
+            "DiaSources.",
+        name="{fakesType}{coaddName}Diff_fullDiaObjTable",
+        storageClass="DataFrame",
+        dimensions=("tract", "patch"),
+    )
+
+
+class DrpDiaCalculationPipeConfig(
+        pipeBase.PipelineTaskConfig,
+        pipelineConnections=DrpDiaCalculationPipeConnections):
+    filterNames = pexConfig.ListField(
+        dtype=str,
+        default=['u', 'g', 'r', 'i', 'z', 'y'],
```

Comment:
See my comment in the ap_association review about whether you could end up with a list of lists.

Reply:
See my response there, the tasks have different context and the list of lists problem won't happen.